Gotrue/NetworkStatus.cs

```diff
@@ -13,58 +15,114 @@ namespace Supabase.Gotrue
 	/// </summary>
 	public class NetworkStatus
 	{
-		private readonly IGotrueClient<User, Session> _client;
+		private readonly List<NetworkListener> _listeners = new List<NetworkListener>();
+
+		/// <summary>
+		/// True if the network has been checked.
+		/// </summary>
+		public bool Ready;
+
+		/// <summary>
+		/// A delegate for listening to network changes.
+		/// </summary>
+		public delegate void NetworkListener(bool isNetworkAvailable);
+
+		/// <summary>
+		/// Adds a listener to the network status system.
+		/// </summary>
+		/// <param name="listener"></param>
+		public void AddListener(NetworkListener listener)
+		{
+			_listeners.Add(listener);
+		}
+
 		/// <summary>
-		/// Set up a listener for the network status.
+		/// Removes a listener from the network status system.
 		/// </summary>
-		/// <param name="client"></param>
-		public NetworkStatus(IGotrueClient<User, Session> client)
+		/// <param name="listener"></param>
+		public void RemoveListener(NetworkListener listener)
 		{
-			_client = client;
+			_listeners.Remove(listener);
+		}
+
+		private void NotifyListeners(bool isNetworkAvailable)
+		{
+			foreach (NetworkListener listener in _listeners)
+			{
+				listener?.Invoke(isNetworkAvailable);
+			}
 		}
 
+		/// <summary>
+		/// The <see cref="Client"/> that this network status system is attached to.
+		/// </summary>
+		public IGotrueClient<User, Session>? Client { get; set; }
+
 		/// <summary>
 		/// Pings the URL in the <see cref="Client.Options"/> to check if the network is online.
+		/// 
+		/// https://wutdruyqartmqvkdaxnf.supabase.co/auth/v1/settings
```

Comment:
Probably want to remove your instance here!

Reply:
lol thanks!